### PR TITLE
Add gptq tutorial for torch

### PR DESCRIPTION
**Pull request by irenaby**
## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+   "execution_count": null,
+   "id": "b3f0acc8-281c-4bca-b0b9-3d7677105f19",
+   "metadata": {
+    "id": "b3f0acc8-281c-4bca-b0b9-3d7677105f19"
+   },
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "from torch.utils.data import DataLoader\n",
+    "from torchvision.models import mobilenet_v2, MobileNet_V2_Weights\n",
```
Comment (ofirgo) [inline tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb:63]: why do you need to import the weights?
can't the model be imported like in this [tutorial](https://github.com/sony/model_optimization/blob/efe8dc9f9d4cfbc6a1254ebe57726fc55916e20e/tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_export.ipynb#L84)?

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+   "source": [
+    "from model_compression_toolkit.gptq.common.gptq_constants import REG_DEFAULT\n",
+    "\n",
+    "# Specify the target platform capability (TPC)\n",
+    "tpc = mct.get_target_platform_capabilities(\"pytorch\", 'imx500', target_platform_version='v1')\n",
+    "\n",
+    "# Create a GPTQ quantization configuration and set the number of training iterations. \n",
+    "# For the sake of running faster, the hessian based weights are disabled in this tutorial.\n",
+    "gptq_config = mct.gptq.get_pytorch_gptq_config(n_epochs=50,\n",
+    "                                               use_hessian_based_weights=False,\n",
```
Comment (ofirgo) [inline tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb:221]: Please check the following:
Set use_hessian_based_weights to True, and run it with the nightly MCT version (change the import just for the sake of this test, don't push it to the actual tutorial).
Monitor how longer is the execution, compared to a run without the Hessians.
If it is not significant (say+-10 minutes), then, set use_hessian_based_weights to True by default in the tutorial (or don't mention it at all because this is the default value anyway) and modify the comment above the code block (say that the execution might take some time)

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2edacb5b7779e4d8",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from model_compression_toolkit.gptq.common.gptq_constants import REG_DEFAULT\n",
```
Comment (ofirgo) [inline tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb:213]: No need to import and pass this constant, it is the default value in the config anyway (I know it is like that in the Keras tutorial, it is not needed there either)

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb (hunk 2 above)
Comment (ofirgo) [reply tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb:221]: Also verify the difference in the results (accuracy)

**Review by ofirgo — CHANGES_REQUESTED** [review]

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb (hunk 1 above)
Comment (irenaby) [reply tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb:63]: This is the way to load pretrained weights. In the tutorial you linked weights are not loaded, in other tutorials the model is loaded with pretrained=Train which is deprecated in torchvision. https://pytorch.org/vision/stable/models.html#initializing-pre-trained-models

Comment (irenaby) [reply tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb:63]: Also the weights object contains the transforms (it's an official API)

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb (hunk 2 above)
Comment (irenaby) [reply tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mobilenet_gptq.ipynb]: Changed. The whole run takes less than 5min as I only use 50 epochs. I'm not sure why in keras tutorial 5000 is used.
